thriftbp/server_middlewares_test.go

```diff
@@ -332,3 +333,76 @@ func TestPanicMiddleware(t *testing.T) {
 		}
 	})
 }
+
+func TestPrometheusMetricsMiddleware(t *testing.T) {
+	t.Run("bp error", func(t *testing.T) {
```

Comment:
nit: you can convert the 4 subtests into a table test. the fields of the table are:
```
struct {
  err error
  next thrift.WrappedTProcessorFunction
  expectedOK bool
  expectedErr error
}
```
but you are not really checking any actual prometheus metrics in the tests. you are just checking that the middleware does not change ok and error values from next call.